src/server/node_discovery.py

```diff
@@ -319,11 +321,11 @@ async def _periodically_cleanup(self):
             full_node_connected = self.server.get_full_node_connections()
             connected = [c.get_peer_info() for c in full_node_connected]
             connected = [c for c in connected if c is not None]
-            if len(connected) >= 3:
+            if self.address_manager is not None and len(connected) >= 3:
```

Comment:
Not 100% sure if this is the expected behavior or we prefer something like `assert self.address_manager is not None`

Reply:
this is fine